src/main/java/aniela/remanent/pdf/report/abstraction/ReportPdf.java

```diff
@@ -0,0 +1,225 @@
+        fillTableWithReportPageNumberAndSum(tableSumamry, reportPagesForSummary, summaryGenerator.getTotalSum(reportPagesForSummary));
+        document.add(tableSumamry);
+    }
+
+    private void closeDocument() {
+        document.close();
+    }
+
+    private void addEmptyLine(Paragraph paragraph, int number) {
+        for (int i = 0; i < number; i++) {
+            paragraph.add(new Paragraph(" "));
+        }
+    }
+
+    private PdfPCell getPdfCellHeader(String columnName) {
+        PdfPCell pdfCell = new PdfPCell(new Phrase(columnName, FONT_HEADER));
+        pdfCell.setHorizontalAlignment(Element.ALIGN_CENTER);
+        return pdfCell;
+    }
+
+    private PdfPCell getPdfCell(String value) {
+        PdfPCell pdfCell = new PdfPCell(new Phrase(value, FONT_VALUE));
+        pdfCell.setHorizontalAlignment(Element.ALIGN_CENTER);
+        return pdfCell;
+    }
+
+    private PdfPCell getPdfCellNazwaTowaru(String value) {
+        PdfPCell pdfCell = new PdfPCell(new Phrase(value, FONT_VALUE));
+        pdfCell.setHorizontalAlignment(Element.ALIGN_LEFT);
+        return pdfCell;
+    }
+
+    private PdfPCell getEmptyPdfCell() {
+        PdfPCell pdfCell = new PdfPCell(new Phrase("", FONT_VALUE));
+        pdfCell.setBorderWidth(0f);
+        return pdfCell;
+    }
+
+    private PdfPTable getPageTable() throws DocumentException {
+        PdfPTable table = new PdfPTable(6);
+        table.setWidthPercentage(100f);
+        table.setTotalWidth(new float[]{10f, 80f, 7f, 10f, 14f, 17f});
+
+        PdfPCell pdfCellHeaderLp = getPdfCellHeader("LP");
+        PdfPCell pdfCellHeaderNazwaTowaru = getPdfCellHeader("Nazwa towaru");
+        PdfPCell pdfCellHeaderJm = getPdfCellHeader("j. m.");
+
+        PdfPCell pdfCellHeaderIlosc = getPdfCellHeader("Ilość");
+        PdfPCell pdfCellHeaderCenaNetto = getPdfCellHeader("Cena netto");
+        PdfPCell pdfCellHeaderwartoścNetto = getPdfCellHeader("Wartość netto");
+        table.addCell(pdfCellHeaderLp);
+        table.addCell(pdfCellHeaderNazwaTowaru);
+        table.addCell(pdfCellHeaderJm);
+        table.addCell(pdfCellHeaderIlosc);
+        table.addCell(pdfCellHeaderCenaNetto);
+        table.addCell(pdfCellHeaderwartoścNetto);
+        table.setHeaderRows(1);
+        return table;
+    }
+
+    private void fillPageWithPostions(PdfPTable table, ReportPage reportPage) {
+        reportPage.positions.forEach(element -> {
+            table.addCell(getPdfCell(String.valueOf(element.getPozyzjaWRaporcie())));
+            table.addCell(getPdfCellNazwaTowaru(element.getNazwaTowaru()));
+            table.addCell(getPdfCell(element.getJednostka()));
+            table.addCell(getPdfCell(AmountExtractor.extractAmount(element.getIlosc())));
+            table.addCell(getPdfCell(element.getCenaNetto() + POLISH_CURRENCY));
+            table.addCell(getPdfCell(element.getSumaNetto() + POLISH_CURRENCY));
+        });
+        table.addCell(getEmptyPdfCell());
```

Comment:
a to co ? robisz puste celki aby na koniec dac sume ?
moze w jaka metode to zawinac i nazwac ?

Reply:
tak, trzeba dostwic puste celki szyba celka z suma wyszla

Reply:
Jezeli wyjasnienie jest OK to zamknij comment albo napisz OK